components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/signup/UserSelfRegistrationManager.java

```diff
@@ -184,9 +184,10 @@ public NotificationResponseBean registerUser(User user, String password, Claim[]
 
         // Provide support for passwordless registration.
         // If the password is mandatory and not provided, it will be failed at the attribute verification.
+        boolean isPasswordPolicyValidationSkippedByDefault = UserCoreUtil.getSkipPasswordPatternValidationThreadLocal();
         if (password == null) {
             UserCoreUtil.setSkipPasswordPatternValidationThreadLocal(true);
```

Comment:
Shall we set this conditionally based on `isPasswordPolicyValidationSkippedByDefault `